packages/bitcore-lib/lib/crypto/schnorr.js

```diff
@@ -0,0 +1,138 @@
+const crypto = require('crypto');
+const $ = require('../util/preconditions');
+const JS = require('../util/js');
+const BN = require('./bn');
+const Point = require('./point');
+const TaggedHash = require('./taggedhash');
+
+const Schnorr = function Schnorr() {
+  if (!(this instanceof Schnorr)) {
+    return new Schnorr();
+  }
+  return this;
+};
+
+Schnorr.prototype.set = function() {};
+
+/**
+ * Create a schnorr signature
+ * @param {PrivateKey|Buffer|BN} privateKey
+ * @param {String|Buffer} message Hex string or buffer
+ * @param {String|Buffer} aux Hex string or buffer
+ * @returns {Buffer}
+ * @link https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#Default_Signing
+ */
+Schnorr.sign = function(privateKey, message, aux) {
+  privateKey = Buffer.isBuffer(privateKey) ? privateKey : privateKey.toBuffer();
+  if (privateKey.length !== 32) {
+    throw new Error('Private key should be 32 bytes for schnorr signatures');
+  }
+
+  if (typeof message === 'string' && JS.isHexaString(message)) {
```

Comment:
What should happen if `typeof message === 'string'` is true, but JS.isHexaString(message) is false?

On the buffer concatenation of the assignment to "rand" on line 55, 'message' is not ensured to be a Buffer.

Reply:
I see - the following line is ensuring that message is a Buffer further in the code. Here's a way that should perform nearly the same number of operations, but provides better IDE type help later in the code:
`  
  // if (typeof message === 'string' && JS.isHexaString(message)) {
  //   message = Buffer.from(message, 'hex')
  // }
  // $.checkArgument($.isType(message, 'Buffer'), 'message must be a hex string or buffer');
  if (typeof message === 'string') {
    $.checkArgument(JS.isHexaString(message), 'message must be a hex string or buffer');
    message = Buffer.from(message, 'hex')
  }
  $.checkArgument($.isType(message, 'Buffer'), 'message must be a hex string or buffer');
`